packages/math-input/src/components/input/math-wrapper.ts

```diff
@@ -156,69 +111,15 @@ class MathWrapper {
     }
 
-    _writeNormalFunction(name: string) {
-        this.mathField.write(`\\${name}\\left(\\right)`);
-        this.mathField.keystroke("Left");
-    }
-
     /**
      * Handle a key press and return the resulting cursor state.
-     *
-     * @param {Key} key - an enum representing the key that was pressed
-     * @returns {object} a cursor object, consisting of a cursor context
      */
-    pressKey(key: string) {
-        const cursor = this.mathField.__controller.cursor;
-
-        if (key in KeyActions) {
-            const {str, fn} = KeyActions[key];
+    pressKey(key: Key): Cursor {
+        const cursor = this.getCursor();
+        const translator = customKeyTranslator[key];
 
-            if (str && fn) {
-                this.mathField[fn](str);
-            }
-        } else if (Object.keys(NormalCommands).includes(key)) {
-            this._writeNormalFunction(NormalCommands[key]);
-        } else if (key === "FRAC_EXCLUSIVE") {
-            // If there's nothing to the left of the cursor, then we want to
-            // leave the cursor to the left of the fraction after creating it.
-            const shouldNavigateLeft = cursor[this.MQ.L] === ActionType.MQ_END;
-            this.mathField.cmd("\\frac");
-            if (shouldNavigateLeft) {
-                this.mathField.keystroke("Left");
-            }
-        } else if (key === "FRAC") {
-            // eslint-disable-next-line @typescript-eslint/no-unused-vars
-            const shouldNavigateLeft = cursor[this.MQ.L] === ActionType.MQ_END;
-            this.mathField.cmd("\\frac");
-        } else if (key === "LOG_N") {
-            this.mathField.write("log_{ }\\left(\\right)");
-            this.mathField.keystroke("Left"); // into parentheses
-            this.mathField.keystroke("Left"); // out of parentheses
-            this.mathField.keystroke("Left"); // into index
-        } else if (key === "CUBE_ROOT") {
-            this.mathField.write("\\sqrt[3]{}");
-            this.mathField.keystroke("Left"); // under the root
-        } else if (key === "EXP" || key === "EXP_2" || key === "EXP_3") {
-            this._handleExponent(cursor, key);
-        } else if (
-            key === "JUMP_OUT_PARENTHESES" ||
-            key === "JUMP_OUT_EXPONENT" ||
-            key === "JUMP_OUT_BASE" ||
-            key === "JUMP_INTO_NUMERATOR" ||
-            key === "JUMP_OUT_NUMERATOR" ||
-            key === "JUMP_OUT_DENOMINATOR"
-        ) {
-            this._handleJumpOut(cursor, key);
-        } else if (key === "BACKSPACE") {
-            this._handleBackspace(cursor);
-        } else if (key === "LEFT") {
-            this._handleLeftArrow(cursor);
-        } else if (key === "RIGHT") {
-            this._handleRightArrow(cursor);
-        } else if (/^[a-zA-Z]$/.test(key)) {
-            this.mathField[ActionType.WRITE](key);
-        } else if (/^NUM_\d/.test(key)) {
-            this.mathField[ActionType.WRITE](key[4]);
+        if (translator) {
```

Comment:
If this is `null`, is that an issue?  Or is it just that we don't support a translation for the keystroke and can just let mathquill handle it directly?  Or are we sweeping a misconfiguration under the rug here?

Reply:
If we don't have a handler for the key pressed, we do nothing. This is how the original code worked too.

Just to be clear though, all of this is translating keypad key presses. It has nothing to do with typing in the input. Just mention that since you said "keystroke."